pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -230,6 +253,24 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
                     }
                 }
             }
+
+            // Update if the markDeletePosition move forward
+            updateIfNeededAndGetLastSentPosition();
+
+            // Should not access to individualDeletedMessages from outside managed cursor
+            // because it doesn't guarantee thread safety.
+            if (lastSentPosition == null) {
+                if (cursor.getMarkDeletedPosition() != null) {
+                    lastSentPosition = ((ManagedCursorImpl) cursor)
+                            .processIndividuallyDeletedMessagesAndGetMarkDeletedPosition(range -> {
+                                final PositionImpl lower = range.lowerEndpoint();
+                                final PositionImpl upper = range.upperEndpoint();
+                                individuallySentPositions.addOpenClosed(lower.getLedgerId(), lower.getEntryId(),
+                                        upper.getLedgerId(), upper.getEntryId());
+                                return true;
+                            });
+                }
+            }
```

Comment:
These initialize only will be called on the following cases:
- `cursor.recover`
- the last consumer is removed.

Maybe we can move these codes to another place.